diff --git a/hpm_sdk/components/enet_phy/rtl8201/hpm_rtl8201.c b/hpm_sdk/components/enet_phy/rtl8201/hpm_rtl8201.c
index 787064d13..05fe9aad3 100644
--- a/hpm_sdk/components/enet_phy/rtl8201/hpm_rtl8201.c
+++ b/hpm_sdk/components/enet_phy/rtl8201/hpm_rtl8201.c
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2021-2025 HPMicro
+ * Copyright (c) 2021-2026 HPMicro
  *
  * SPDX-License-Identifier: BSD-3-Clause
  *
@@ -89,13 +89,13 @@ bool rtl8201_basic_mode_init(ENET_Type *ptr, uint32_t phy_addr, rtl8201_config_t
 
     enet_write_phy(ptr, phy_addr, RTL8201_BMCR, data);
 
-#if defined(RMII) && RMII
-    if (config->rmii_refclk_dir == enet_phy_rmii_refclk_dir_in) {
-        rtl8201_set_rmii_refclk_direction(ptr, phy_addr, rtl8201_config_refclk_input);
-    } else {
-        rtl8201_set_rmii_refclk_direction(ptr, phy_addr, rtl8201_config_refclk_output);
+    if (config->media_interface == enet_inf_rmii) {
+        if (config->rmii_refclk_dir == enet_phy_rmii_refclk_dir_in) {
+            rtl8201_set_rmii_refclk_direction(ptr, phy_addr, rtl8201_config_refclk_input);
+        } else {
+            rtl8201_set_rmii_refclk_direction(ptr, phy_addr, rtl8201_config_refclk_output);
+        }
     }
-#endif
 
     return true;
 }
@@ -131,4 +131,4 @@ void rtl8201_set_rmii_refclk_direction(ENET_Type *ptr, uint32_t phy_addr, uint8_
 
     /* write register RMSR */
     enet_write_phy(ptr, phy_addr, RTL8201_RMSR_P7, data);
-}
\ No newline at end of file
+}
diff --git a/hpm_sdk/components/enet_phy/rtl8201/hpm_rtl8201.h b/hpm_sdk/components/enet_phy/rtl8201/hpm_rtl8201.h
index 80280a748..2902a42ed 100644
--- a/hpm_sdk/components/enet_phy/rtl8201/hpm_rtl8201.h
+++ b/hpm_sdk/components/enet_phy/rtl8201/hpm_rtl8201.h
@@ -6,7 +6,7 @@
  */
 
 /*
- * Copyright (c) 2021-2025 HPMicro
+ * Copyright (c) 2021-2026 HPMicro
  *
  * SPDX-License-Identifier: BSD-3-Clause
  *
@@ -66,7 +66,8 @@ typedef struct {
     uint8_t speed;              /**< Port speed: @ref enet_phy_port_speed_t */
     bool auto_negotiation;      /**< Enable/disable auto-negotiation */
     uint8_t duplex;             /**< Duplex mode: @ref enet_phy_duplex_mode_t */
-    uint8_t rmii_refclk_dir;   /**< RMII reference clock direction: @ref rtl8201_config_refclk_direction_t */
+    uint8_t rmii_refclk_dir;    /**< RMII reference clock direction: @ref rtl8201_config_refclk_direction_t */
+    uint8_t media_interface;    /**< Interface type: @ref enet_inf_type_t */
 } rtl8201_config_t;
 
 #if defined(__cplusplus)
